src/region.rs

```diff
@@ -39,6 +40,20 @@ impl RegionWithLeader {
         key >= start_key && (key < end_key || end_key.is_empty())
     }
 
+    pub fn contains_start_bound(&self, bound: Bound<&Key>) -> bool {
+        let start_key = self.region.get_start_key();
+        let end_key = self.region.get_end_key();
+
+        match bound {
+            Bound::Included(key) => self.contains(key),
+            Bound::Excluded(key) => {
+                let key: &[u8] = key.into();
+                key >= start_key && (key <= end_key || end_key.is_empty())
```

Comment:
I don't get the difference between `contains` and `contains_start_bound`.

```
key space: ---------------------
        region: [_____)
               range: (_______
```

The end key of the region is excluded. If the given start bound is also excluded, when they equal, the region doesn't seem to _contain_ any part of the given range (so it seems to me that we should not say it contains the start bound).

Reply:
Emm... I change to `(Vec<u8>, Vec<u8>)` as it seems that `BoundRange` is not necessary.
`contains_start_bound` is useless now and I remove it.